torch/_jit_internal.py

```diff
@@ -151,6 +154,29 @@ def fn(*args, **kwargs):
     return fn
 
 
+def overload(self_type, fn_name, fns):
```

Comment:
I feel like this logic should be part of ScriptModule and not a separate thing, especially because it has to be done per instance of a script module.

Reply:
Should we keep this as an internal thing with `self._overload(...)`? Or expose it `self.overload(...)`?

Reply:
We can  do `_overload` for now. Another API option is:

```

class ScriptModule:
  my_function = torch.jit._overload('my_function_tuple', 'my_function_tensor')
```
This would eliminate the need to call something in the constructor.